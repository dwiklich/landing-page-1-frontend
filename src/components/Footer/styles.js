import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionComponent } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionComponent} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`;
