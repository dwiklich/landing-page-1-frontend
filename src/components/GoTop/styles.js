import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
