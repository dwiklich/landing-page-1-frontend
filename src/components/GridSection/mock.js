export default {
  background: false,
  title: 'My grid',
  description: 'Lorem ipsum dolor sit,',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Repellat iure porro in autem deserunt quis, voluptas hic est recusandae veniam modi, assumenda molestias et at eveniet vero eum error cumque?',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Repellat iure porro in autem deserunt quis, voluptas hic est recusandae veniam modi, assumenda molestias et at eveniet vero eum error cumque?',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Repellat iure porro in autem deserunt quis, voluptas hic est recusandae veniam modi, assumenda molestias et at eveniet vero eum error cumque?',
    },
  ],
};
