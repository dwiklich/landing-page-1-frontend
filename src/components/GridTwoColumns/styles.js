import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;

    > img {
      width: 50rem;
    }
  `}
`;
