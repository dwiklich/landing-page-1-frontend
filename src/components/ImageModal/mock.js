export default {
  imageModal: ['https://source.unsplash.com/random/800x800?r=4', 'Teste 4'],
};
