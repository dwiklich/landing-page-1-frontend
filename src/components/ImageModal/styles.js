import styled, { css } from 'styled-components';

/* The Modal (background) */
export const Container = styled.div`
  ${({ theme, modal }) => css`
    display: ${modal ? 'none' : 'block'}; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%; /* Full width */
    height: 100%; /* Full height */
    overflow: auto; /* Enable scroll if needed */
    background-color: rgb(0, 0, 0); /* Fallback color */
    background-color: rgba(0, 0, 0, 0.4); /* Black w/ opacity */
  `}
`;

/* Modal Content */
export const Content = styled.div`
  ${({ theme, modal }) => css`
    position: absolute;
    left: 50%;
    bottom: 50%;
    transform: translateX(-50%) translateY(50%);

    background-color: #fefefe;
    margin: auto;
    padding: 20px;
    border: 1px solid #888;
    width: 80%;

    @media ${theme.media.lteMedium} {
      width: 100%;
    }
  `}
`;

/* Modal Button */
export const ButtonClose = styled.span`
  ${({ theme, modal }) => css`
    color: #aaaaaa;
    float: right;
    font-size: 28px;
    font-weight: bold;

    &:hover,
    &:focus {
      color: #000;
      text-decoration: none;
      cursor: pointer;
    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    position: relative;
    left: 50%;
    transform: translateX(-50%);

    background-size: cover;
    width: 60%;
    border-radius: 1rem;
  `}
`;
