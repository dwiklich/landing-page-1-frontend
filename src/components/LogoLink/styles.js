import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme, srcImage }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    & > img {
      max-height: 3rem;
    }
  `}
`;
