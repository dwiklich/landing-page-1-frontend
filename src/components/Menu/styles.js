import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    background: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      /* display: ${visible ? 'flex' : 'none'}; */

      visibility: hidden;
      opacity: 0;

      ${visible && menuVisible(theme)}

      height: 100vh;

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: scroll;

        &::-webkit-scrollbar {
          display: none;
        }
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      /* flex-direction: column; */
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    height: 4rem;
    width: 4rem;
    background: ${visible ? theme.colors.white : theme.colors.primaryColor};
    color: ${visible ? theme.colors.primaryColor : theme.colors.white};
    border: none;
    border-radius: 0.5rem;
    display: none;

    @media ${theme.media.lteMedium} {
      ${visible && menuVisible(theme)}

      display: flex;
      justify-content: center;
      align-items: center;

      > svg {
        width: 2.5rem;
        height: 2.5rem;
      }
    }
  `}
`;
