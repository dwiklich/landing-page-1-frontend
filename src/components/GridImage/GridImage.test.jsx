import { screen, fireEvent } from '@testing-library/react';

import { renderTheme } from '../../styles/render-theme';
import { GridImage } from '.';
import mock from './mock';

describe('<GridImage />', () => {
  it('should setModal true with click element image', () => {
    const { container } = renderTheme(<GridImage {...mock} />);

    const image = screen.getByRole('img', { name: 'Teste 1' });

    fireEvent.click(image);
    expect(image).toBeInTheDocument();
  });

  it('should render not background', () => {
    const { container } = renderTheme(
      <GridImage
        title={mock.title}
        description={mock.description}
        grid={mock.grid}
      />,
    );

    expect(container).toBeInTheDocument();
  });
  it('should render', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toBeInTheDocument();
  });
  it('should Open and Close image of grid ', () => {
    const { container } = renderTheme(<GridImage {...mock} />);

    const image = screen.getByRole('img', { name: 'Teste 5' });
    const modal = screen.getByLabelText('modal');
    const button = screen.getByLabelText('modal-button-close');

    fireEvent.click(image);
    expect(modal).toHaveStyleRule('display', 'block');

    fireEvent.click(button);
    expect(modal).toHaveStyleRule('display', 'none');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
