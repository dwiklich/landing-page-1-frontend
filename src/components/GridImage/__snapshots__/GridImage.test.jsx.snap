// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should match snapshot 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  display: none;
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0,0,0);
  background-color: rgba(0,0,0,0.4);
}

.c10 {
  position: absolute;
  left: 50%;
  bottom: 50%;
  -webkit-transform: translateX(-50%) translateY(50%);
  -ms-transform: translateX(-50%) translateY(50%);
  transform: translateX(-50%) translateY(50%);
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
}

.c11 {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.c11:hover,
.c11:focus {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c12 {
  position: relative;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-size: cover;
  width: 60%;
  border-radius: 1rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  background-size: cover;
  width: 100%;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c10 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: repeat(auto-fill,minmax(200px,1fr));
  }
}

<section
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        My grid
      </h1>
      <div
        class="c4 c5"
      >
        Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Teste 1"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=1"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 2"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=2"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 3"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=3"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 4"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=4"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 5"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=5"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 6"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=6"
          />
        </div>
      </div>
      <div
        aria-label="modal"
        class="c9"
      >
        <div
          class="c10"
        >
          <span
            aria-label="modal-button-close"
            class="c11"
          >
            Ã—
          </span>
          <img
            aria-label="image-modal"
            class="c12"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
